### PR TITLE
Add basic internal testing for memory allocations

### PR DESCRIPTION
**Pull request by joelhawksley**
### What are you trying to accomplish?

This PR adds basic internal testing tooling for memory allocations, giving us a starting point for optimizing memory usage in the project.

### What approach did you choose and why?

I copied the `assert_allocations` helper from Primer ViewComponents, but modified it to target specific Ruby versions to remove ambiguity.

### CODE REVIEW

**Comment by reeganviljoen** [comment]
@joelhawksley many production set-ups use jemalloc(its even in the default dockerfile for rails) so it may be usefull to test against it aswell

**Comment by joelhawksley** [comment]
@reeganviljoen hmm. I'd worry about forcing folks to set up jemalloc locally in their dev environment to work against these tests. Do you think that'd be too much to ask?

File: test/sandbox/test/rendering_test.rb [hunk 1]
```diff
@@ -9,6 +9,19 @@ def test_render_inline
     assert_selector("div", text: "hello,world!")
   end
 
+  def test_render_inline_allocations
+    # Stabilize compilation status ahead of testing allocations to simulate rendering
+    # performance with compiled component
+    ViewComponent::CompileCache.cache.delete(MyComponent)
+    MyComponent.ensure_compiled
+
+    assert_allocations("3.4.0" => 107, "3.3.5" => 116, "3.3.0" => 129, "3.2.5" => 115, "3.1.6" => 115, "3.0.7" => 125) do
```
Comment (camertron) [inline test/sandbox/test/rendering_test.rb:18]: Wow, pretty cool there's a ~15% decrease in allocations from Ruby 3.0 to 3.4 🎉

**Review by camertron — APPROVED** [review]
Awesome!

**Pull request merged**